atc/exec/check_step.go

```diff
@@ -337,13 +345,9 @@ func (step *CheckStep) runCheck(
 	)
 }
 
-func (step *CheckStep) containerOwner(resourceConfig db.ResourceConfig) db.ContainerOwner {
-	if step.plan.Resource == "" {
-		return db.NewBuildStepContainerOwner(
-			step.metadata.BuildID,
-			step.planID,
-			step.metadata.TeamID,
-		)
+func (step *CheckStep) containerOwner(delegate CheckDelegate, resourceConfig db.ResourceConfig) db.ContainerOwner {
+	if !step.plan.IsPeriodic() {
```

Comment:
What's this change for? Doesn't this mean that checks for resource types (and prototypes) will now use the resource config check session container owner?

Reply:
Good catch! I made the change in the very beginning of this PR. At the time, I hadn't added `NewInMemoryCheckBuildContainerOwner`. I can revert this change. But to confirm, is the point that we don't resource type check to reuse container? @aoldershaw